osquery/tables/system/posix/sudoers.cpp

```diff
@@ -0,0 +1,102 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <locale>
+#include <vector>
+
+#include <boost/algorithm/string/trim.hpp>
+
+#include <osquery/filesystem.h>
+#include <osquery/tables.h>
+
+#include "osquery/core/conversions.h"
+
+namespace osquery {
+namespace tables {
+
+const std::string kSudoFile = "/etc/sudoers";
+
+std::vector<std::string> readFromFile(const std::string& path) {
+  std::vector<std::string> sudo_lines;
+
+  if (!isReadable(path).ok()) {
+    return sudo_lines;
+  }
+
+  std::string contents;
+  if (!forensicReadFile(path, contents).ok()) {
+    return sudo_lines;
+  }
+
+  auto lines = split(contents, "\n");
+
+  for (auto& line : lines) {
+    boost::trim(line);
+
+    // Only add lines that are not comments or blank.
+    if (line.size() > 0 && line.at(0) != '#') {
+      sudo_lines.push_back(line);
+    }
+  }
+
+  return sudo_lines;
+}
+
+Row genSudoRule(const std::string& line) {
+  Row r;
+
+  int space = 0;
+  while (space < (int)line.length() && !std::isspace(line.at(space))) {
+    ++space;
+  }
+
+  if (space >= (int)line.length()) {
+    // Whitespace not found, faulty line.
+    return r;
+  }
+
+  r["header"] = line.substr(0, space);
+  r["rule_details"] = line.substr(space + 1, std::string::npos);
+
+  return r;
+}
+
+QueryData genSudoers(QueryContext& context) {
+  QueryData results;
+
+  auto lines = readFromFile(kSudoFile);
+
+  QueryData defaults;
```

Comment:
These objects were used in order to have a loose ordering of the entries in the resultant table (i.e. all "Defaults" before "(Users)" before "(Groups)", regardless of their order in the original sudoers file). If this ordering isn't deemed important, we can remove these objects and just push everything to "results" as you suggested.